lib/kubernetes-deploy/kubernetes_resource/job.rb

```diff
@@ -30,8 +31,20 @@ def status
       end
     end
 
+    def failure_message
+      if (condition = failed_status_condition.presence)
+        "#{condition['reason']} (#{condition['message']})"
+      end
+    end
+
     private
 
+    def failed_status_condition
+      (@instance_data.dig("status", "conditions") || []).detect do |condition|
```

Comment:
nit: this would be easier to read if it used a guard clause instead of the `|| []` trick. I personally prefer `find` over `detect`, but I see we already use both in this repo.